src/goto-programs/initialize_goto_model.cpp

```diff
@@ -133,10 +133,36 @@ goto_modelt initialize_goto_model(
   }
   else if(!binaries_provided_start)
   {
-    // Allow all language front-ends to try to provide the user-specified
-    // (--function) entry-point, or some language-specific default:
-    entry_point_generation_failed=
-      language_files.generate_support_functions(goto_model.symbol_table);
+    if(options.is_set("function"))
+    {
+      // no entry point is present; Use the mode of the specified entry function
+      // to generate one
+      auto const &entry_function_name = options.get_option("function");
+      auto const entry_function_sym =
+        goto_model.symbol_table.lookup({entry_function_name});
```

Comment:
Why the braces?

Reply:
It's calling the `irep_idt` constructor

Reply:
But ... you don't need to do that, the compiler will happily do it for you?

Reply:
Yes, but I don't particularly like implicit conversions.

Reply:
Agreed, and I think this is only one of two implicit conversions that we permit (the other being from integer types to `mp_integer`). But changing the implicit to an explicit conversion only once will not add clarity, and even less so when 1) it's done without naming the type and 2) from an `auto`-typed variable.

But also: this is me doing my usual nit-picking.